### ISSUE TITLE
docs: Refactor documented functions to be methods rather than properties

### ISSUE THREAD

**Issue by maschad** [libp2p/js-libp2p#2096]
In order for methods to not be rendered as properties on the API docs e.g. the [Libp2p Event Docs](https://libp2p.github.io/js-libp2p/interfaces/_libp2p_interface.index.Libp2pEvents.html) they should be refactored as such.

e.g.

```ts
/**
 * @packageDocumentation
 *
 * Exports a `Libp2p` type for modules to use as a type argument.
 *
 * @example
 *
 * ```typescript
 * import type { Libp2p } from '@libp2p/interface'
 *
 * function doSomethingWithLibp2p (node: Libp2p) {
 *   // ...
 * }
 * ```
 */
interface MyInterface {
  /**
   * This is a method description.
   */
  myMethod(arg1: string, arg2: number): boolean
}
```

as opposed to 

```ts
interface MyInterface {
  /**
   * This is a method description.
   */
  myMethod: (arg1: string, arg2: number) => boolean
}

```

**Comment by wemeetagain** [comment]
This should be enforced when we upgrade aegir.
I have a local branch, but got distracted with #2095

**Comment by DarhkVoyd** [comment]
@maschad Hey, is this issue still valid? I'm eager to help with this, I'm new to this project so please let me know how I can get started. Thanks!

**Comment by wemeetagain** [comment]
@DarhkVoyd feel free to take on this issue.
You should be able to resolve this in an automated way via a new linter rule introduced in aegir v41.0.0
(After you update the aegir dependency across the monorepo, you should be able to run `npm run lint -- -- --fix` from the project root to apply linter fixes everywhere)

**Comment by DarhkVoyd** [comment]
@wemeetagain Hey, after updating aegir to 41.0.0 and running `npm run lint -- -- --fix`, the terminal keeps on throwing linting error one after fixing another all over the codebase such as `@typescript-eslint/strict-boolean-expressions` and `@typescript-eslint/await-thenable`. So, should I keep on resolving all these errors along the way until the lint runs smoothly or am I doing something wrong?

**Comment by wemeetagain** [comment]
Yup, keep resolving all the errors you run into. You may have to resolve the linter errors manually.

**Comment by maschad** [comment]
@DarhkVoyd the latest master should help make things a bit easier in your code editor since the [tsconfig is now adapted](https://github.com/libp2p/js-libp2p/pull/2109) for the monorepo

**Comment by DarhkVoyd** [comment]
@wemeetagain @maschad I might need some help as I am unable to understand if I am doing it right and working in the right direction. Kindly review my changes in the PR.

**Comment by achingbrain** [comment]
Fixed by #2132
